File: 242-valid-anagram/242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length())    return false;
        int[] freq = new int[26];
        for(char ch : s.toCharArray()){
            freq[ch - 'a']++;
        }
        for(char ch : t.toCharArray()){
            if(freq[ch-'a'] == 0) return false;
            freq[ch - 'a']--;
        }
        for(int i=0; i < 26;i++){
            if(freq[i] != 0){
                return false;
            }
        }
        return true;
        
    }
}

File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char,int> match;
        for(auto ch : s){
            match[ch]++;
        }
        for(auto ch : t){
            match[ch]--;
            if(match[ch] < 0){
                return false;
            }
        }
        for(auto m : match){
            if(m.second !=0){
                return false;
            }
        }
        return true;
    }
};